src/test/java/rueppellii/backend2/tribes/gameUtility/timeService/TimeServiceTest.java

```diff
@@ -0,0 +1,118 @@
+package rueppellii.backend2.tribes.gameUtility.timeService;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.MockitoAnnotations;
+
+import static rueppellii.backend2.tribes.gameUtility.timeService.TimeConstants.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+class TimeServiceTest {
+
+    @InjectMocks
+    private TimeService timeService;
+
+    private Long calculationSystemTime = System.currentTimeMillis();
+    private Integer buildingLevel;
+    private Integer townhallLevel;
+    private Integer troopLevel;
+    private Double troopUpgradeTimeMultiplier;
+
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    void calculateTimeOfBuildingCreationTest() {
+        assertEquals(calculationSystemTime + BUILDING_PROGRESSION_TIME,
+                calculationSystemTime + 30000L);
+    }
+
+    @Test
+    void calculateTimeDifferenceOfBuildingUpgradeFromLevelOneAndTwoTest() {
+        buildingLevel = 1;
+        townhallLevel = 1;
+
+        Long TownHallLevelOneAndBuildingLevelOne = calculationSystemTime +
+                (BUILDING_PROGRESSION_TIME * buildingLevel) / townhallLevel;
+
+        buildingLevel = 2;
+        Long TownHallLevelOneAndBuildingLevelTwo = calculationSystemTime +
+                (BUILDING_PROGRESSION_TIME * buildingLevel) / townhallLevel;
```

Comment:
calculationSystemTime +   (BUILDING_PROGRESSION_TIME * buildingLevel) / townhallLevel

this line could be separated in a different method, because you use it so many times. That method might have 2 input parameters, and you can add it like: buildingTime(1,1); so you dont have to add new value to your variables

Reply:
I modified the class and from now I cal timeService methods directly. Rearranged the variables and put a few in BeforeEach setUp function, however, I feel that those are not necessary there. I change the content of the class via github editor